concent_api/api_testing_helpers.py

```diff
@@ -73,7 +74,10 @@ def api_request(host, endpoint, private_key, public_key, data = None, headers =
         )
         print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
         print('MESSAGE:')
-        print_golem_message(decoded_response, private_key, public_key)
+        if response.headers['Content-Type'] == 'application/json':
+            print(json.loads(response.content.decode())['error'])
```

Comment:
OK. Now you're not handling `application/octet-stream` again :) We're going in circles here.

Please take a step back and think about what we want to achieve here: you're getting a response from the server. It might come from any endpoint: `send/` `receive/`, maybe another one someone adds in the future. The operation might fail (2xx), return a client error (4xx) or a server error (5xx). In case of a success you may get an empty response or a Golem message. You may also get something else if the server is broken and returns 200 with something unexpected. In case of a failure you're likely to get JSON but you can also easily get HTML from nginx if the server crashes completely. You can also get something completely different if the server is misconfigured. This is a test script after all so you cannot assume that the server is working properly.

How to efficiently handle all of this? 
- First assume that the server is giving you a valid response and try to decode it according to what it says. So if it says the content type is JSON, decode it as JSON. If it says `application/octet-steam`, assume a Golem message. If you support any other type, handle it too. If there's no content type, assume that the body is empty. In any other case don't decode and just print what you got, possibly saying that you got an unexpected content type.
- Second, if the decoding method that matches the content type fails or if you can't handle the decoded data (e.g. there's no content type but body is non-empty), catch the error and print an error message saying that you could not decode the response. Then just print the response as is so that the user of the script can try to debug the problem.